Clean up unused variable, assignment flow, etc in buildComponentTemplate

jayphelps opened the pull request:
Mostly flow cleanup and `createdElementBlock` was actually superfluous.

As I'm digging through HTMLBars/Glimmer changes, I'll be PR'ing a number of clean ups. Please let me know if these are unwanted, encouraged or somewhere in between. I'm getting _real_ deep into the changes but I don't want to annoy you all.

Using destructuring since the [style guide says I should](https://github.com/emberjs/ember.js/blob/6808d3b0819f8005391a6a4edce2e9cfbb7fb32f/STYLEGUIDE.md#destructuring)

jayphelps committed:
Clean up unused variable, assignment flow, etc in buildComponentTemplate

rwjblue commented:
Doesn't look like JSCS (possibly just our version) likes the default arguments.

mixonic commented on packages/ember-views/lib/system/build-component-template.js:
```diff
@@ -29,17 +24,13 @@ export default function buildComponentTemplate(componentInfo, attrs, content) {
   }
 
-  return { createdElement: false, block: blockToRender };
+  return { createdElement: !!tagName, block: blockToRender };
```
If you are back in this code, can you add a comment before this line? `tagName` is tricky and though the logic here is correct it is pretty obtuse. Something like:

```js
// tagName is one of:
//
//   * `undefined` if no component is present
//   * the falsy value "" if set explicitly on the component
//   * an actual tagName set explicitly on the component
```

jayphelps replied:
Yip yip

jayphelps commented:
* Removed destructuring default, [not supported yet upstream in esprima](https://github.com/jquery/esprima/issues/1028).
* Added [requested `tagName` comment](https://github.com/emberjs/ember.js/pull/11055#discussion_r29852434).

I dunno what's up with IE9 on saucelabs.

![image](https://cloud.githubusercontent.com/assets/762949/7523388/42108d44-f4b0-11e4-99bd-c01fbdaebb17.png)

rwjblue commented:
Restarted sauce jobs.

rwjblue commented:
Thank you!
